pkg/service/utils.go

```diff
@@ -0,0 +1,110 @@
+package service
+
+import (
+	"io/ioutil"
+	"strings"
+
+	"github.com/coreos/go-systemd/v22/unit"
+)
+
+// CompareServices compare 2 service and return true if serviceA has all the fields of serviceB
+func CompareServices(serviceA, serviceB *Service) (bool, error) {
+	optsA, err := unit.Deserialize(strings.NewReader(serviceA.Content))
+	if err != nil {
+		return false, err
+	}
+	optsB, err := unit.Deserialize(strings.NewReader(serviceB.Content))
+	if err != nil {
+		return false, err
+	}
+
+	for _, optB := range optsB {
+		found := false
+		for _, optA := range optsA {
+			if optA.Match(optB) {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+// CompareServices compare 2 service and return true if serviceA has all the fields of serviceB
+func RemoveFromService(service *Service, options ...*unit.UnitOption) (*Service, error) {
+	opts, err := unit.Deserialize(strings.NewReader(service.Content))
+	if err != nil {
+		return nil, err
+	}
+
+	var newServiceOptions []*unit.UnitOption
+	for _, opt := range opts {
+		found := false
+		for _, optRemove := range options {
+			if opt.Match(optRemove) {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			newServiceOptions = append(newServiceOptions, opt)
+		}
+	}
+
+	data, err := ioutil.ReadAll(unit.Serialize(newServiceOptions))
+	if err != nil {
+		return nil, err
+	}
+
+	return &Service{
+		Name:    service.Name,
+		Path:    service.Path,
+		Content: string(data),
+	}, nil
+}
+
+// ServiceAppend appends missing fields of serviceB into serviceA
+func ServiceAppend(serviceA, serviceB *Service) (*Service, error) {
```

Comment:
I think you can change to  ServiceAppend(serviceA *Service, contentToAppend)((*Service, error)
Why do you need the serviceB?

Reply:
Also maybe change to AppendToService to be same as RemoveFromService